vm/src/api/de.rs

```diff
@@ -856,6 +856,38 @@ impl<'de, 'a, 't> VariantAccess<'de> for Enum<'a, 'de, 't> {
     where
         V: Visitor<'de>,
     {
-        de::Deserializer::deserialize_seq(self.de, visitor)
+        let typ = resolve::remove_aliases_cow(self.de.state.env, &mut NullInterner, self.de.typ);
+        match (self.de.input.as_ref(), &**typ) {
+            (ValueRef::Data(data), &Type::Variant(ref row)) => {
+                match row.row_iter().nth(data.tag() as usize) {
+                    Some(field) => {
+                        let typ = resolve::remove_aliases_cow(
+                            self.de.state.env, &mut NullInterner, &field.typ);
+                        match (data.get_variant(0).unwrap().as_ref(), &**typ) {
```

Comment:
Could Avoid the `unwrap` here and fold the `None` case into "Unable to deserialize"